### PR TITLE
feat(rules): add tally/labels/prefer-stable-order

### PR DESCRIPTION
**Pull request by tinovyatkin**
## Summary

- New rule `tally/labels/prefer-stable-order` reorders LABEL key/value pairs into a deterministic, human-readable order. Default `oci-logical` mode clusters keys by purpose (identity, source/refs, ownership/legal, release/provenance, base image, ecosystem, legacy); `lexical` mode sorts alphabetically.
- Auto-fix emits **narrow per-pair source-region edits** so it cooperates with `tally/labels/prefer-grouped` (FixPriority 96) and `tally/newline-per-chained-call` (97) without overlapping their structural rewrites. Single-line multi-pair LABELs and comment-split blocks are reported only — the splitter handles them first.
- Detection skips when the stage has duplicate keys (defers to `no-duplicate-keys`), when any pair is dynamic / `NoDelim` / has an expansion error, and when an `oci-logical` block contains only unknown reverse-DNS keys with `sort-unknown=false`.
- Implements rollout-plan item 9 from `design-docs/40-label-rules-research.md`.

## Test plan

- [x] `go test ./internal/rules/tally/labels/...` passes (22 unit tests + table-driven cases + idempotency check).
- [x] Coverage on `prefer_stable_order.go` is **87.8%** of statements (above the 85% target).
- [x] `go test ./internal/integration/...` passes — primary fix fixture covers swap, deferred-no-fix, duplicate-key skip, and clean cases.
- [x] Combined-rule fixture (`labels-prefer-stable-order-with-grouped`) demonstrates non-overlapping coordination with `prefer-grouped` and `newline-per-chained-call`.
- [x] `make lint` clean (0 issues).
- [x] `make test` passes (8404+ tests).
- [x] Schemas regenerated via `make schema-gen`; `internal/rules/tally/index.schema.json`, `internal/schemas/registry_gen.go`, `internal/schemas/generated/...`, and `schema.json` updated.
- [x] Docs page added at `_docs/rules/tally/labels/prefer-stable-order.mdx` and registered in `_docs/docs.json` Labels group.

## Summary by CodeRabbit

* **New Features**
  * Added a rule enforcing stable, deterministic ordering of multi-pair Docker LABELs with selectable modes (oci-logical or lexical), optional handling for unknown reverse-DNS keys, and safe auto-fix support.

* **Documentation**
  * Added user-facing docs describing the rule, ordering modes, auto-fix behavior, suppression cases, and configuration options.

* **Tests**
  * Added extensive tests covering behavior, fix generation, edge cases, and idempotency.

* **Chores**
  * Wired rule schema into manifests and registry.

[Review Change Stack](https://app.coderabbit.ai/change-stack/wharflab/tally/pull/715?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: internal/rules/tally/labels/prefer_stable_order_test.go [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+	content := `FROM alpine:3.20
+LABEL org.opencontainers.image.description="d" \
+      org.opencontainers.image.title="t"
+`
+	input := testutil.MakeLintInput(t, "Dockerfile", content)
+	violations := NewPreferStableOrderRule().Check(input)
+	if len(violations) != 1 || violations[0].PreferredFix() == nil {
+		t.Fatalf("baseline case should have a fix")
+	}
+}
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/labels/prefer_stable_order_test.go:632]: ✅ Addressed in 7bcafd34. Thanks @coderabbitai!

File: internal/rules/tally/labels/prefer_stable_order.go [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+	ranks := make([]orderRank, len(inst.Pairs))
+	for i, p := range inst.Pairs {
+		ranks[i] = keyRank(p.Key, i, ctx.mode, ctx.sortUnknown)
+	}
+
+	if ctx.mode == preferStableOrderOCILogical && !ctx.sortUnknown {
+		allUnknown := true
+		for _, r := range ranks {
+			if r.groupRank < unknownReverseDNSGroupRank {
+				allUnknown = false
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/labels/prefer_stable_order.go]: ✅ Addressed in c8997729. Thanks @chatgpt-codex-connector!

File: _docs/rules/tally/labels/prefer-stable-order.mdx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+| Option | Type | Default | Description |
+|--------|------|---------|-------------|
+| `order` | `"oci-logical"` &#124; `"lexical"` | `"oci-logical"` | Comparator. `"oci-logical"` clusters keys by purpose. `"lexical"` sorts purely alphabetically. |
+| `sort-unknown` | boolean | `false` | When `true`, custom reverse-DNS keys (groups 9 and below) are clustered by namespace and sorted lexically within each namespace. When `false`, custom keys keep their relative source order. |
+
+## Related Rules
+
+- [`tally/labels/no-duplicate-keys`](/rules/tally/labels/no-duplicate-keys) —
+  must run first when a duplicate would otherwise change the effective image
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply _docs/rules/tally/labels/prefer-stable-order.mdx:101]: ✅ Addressed in 727902e5. Thanks @greptile-apps!

File: internal/rules/tally/labels/prefer_stable_order.go [hunk 4]
```diff
@@ -0,0 +1,531 @@
...
+	releaseGroupRank            = 4
+	baseImageGroupRank          = 5
+	openshiftCatalogGroupRank   = 6
+	dockerEcosystemGroupRank    = 7
+	legacyGroupRank             = 8
+	unknownReverseDNSGroupRank  = 9
+	unknownUnqualifiedGroupRank = 10
+	dockerExtensionPrefix       = "com.docker.extension."
+	dockerImageSourceEntrypoint = "com.docker.image.source.entrypoint"
+	labelSchemaPrefix           = "org.label-schema."
```
[the remark this answers is not in the document]
Comment (tinovyatkin) [reply internal/rules/tally/labels/prefer_stable_order.go:235]: ✅ Addressed in fe95783f. Thanks @greptile-apps!
